src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -49,5 +55,37 @@ public JoinableTaskFactory JoinableTaskFactory
         {
             get;
         }
+
+        public JoinableTaskCollection ShutdownBlockingTasks { get; }
+
+        private JoinableTaskFactory ShutdownBlockingTaskFactory { get; }
+
+        public CancellationToken DisposalToken => _disposalTokenSource.Token;
+
+        public JoinableTask RunWithShutdownBlockAsync(Func<CancellationToken, Task> func)
+        {
+            return ShutdownBlockingTaskFactory.RunAsync(() => func(DisposalToken));
+        }
+
+        public void Dispose()
+        {
+            // https://github.com/Microsoft/vs-threading/blob/master/doc/cookbook_vs.md#how-to-write-a-fire-and-forget-method-responsibly
+            _disposalTokenSource.Cancel();
```

Comment:
i don't actually understand how the comment applies to this line.

Reply:
It applies to the method implementation, starting with this line.